sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -370,8 +370,11 @@ abstract class CSVSuite
           .load(testFile(carsFile)).collect()
       }
 
-      assert(exception.getMessage.contains("Malformed CSV record"))
-      assert(ExceptionUtils.getRootCause(exception).isInstanceOf[RuntimeException])
+      checkError(
+        exception = ExceptionUtils.getRootCause(exception).asInstanceOf[SparkRuntimeException],
+        errorClass = "MALFORMED_CSV_RECORD",
```

Comment:
hmm... there is `_LEGACY_ERROR_TEMP_2177` on the error chain, should we handle it in this one together?

Reply:
Just added test case for `_LEGACY_ERROR_TEMP_2177`.
Maybe we can assign the proper name for it in the separate ticket